### DISCUSSION TITLE
Shelly Plug S with Tasmota

### DISCUSSION THREAD

**Discussion by ReggiePark** [arendst/Tasmota#13951] [Q&A] [answered]
Hi everyone, new here...

I just got a couple of Shelly-Plug-S devices and the original firmware is not bad. However, you don't seem to be able to calibrate them, so I thought I'd try Tasmota. I haven't actually tried the calibration yet but noticed a couple of other things:

1. I used two devices, one with Tasmota, the other with the original firmware, over a period of 8 hours. Both were sending MQTT messages every 30 seconds and were fairly close to each other with a small load of 1.5 watts for the first two hours, after that zero. The temperature readings from the original firmware device was pretty much flat but the Tasmota device displayed a weird curve:
![image](https://user-images.githubusercontent.com/16917371/145095731-7242da54-bb3e-4980-8fba-e1d6f076d1dc.png)

What's going on there?

2. when I use the http api with the original Shelly to get the status (h..p://ip/status), I get some 800 bytes in 24 ms. If I do this with the Tasmote device (h..p://ip/cm?cmnd=Status%2010) I get some 500 byte in 240 ms. Why does this take so long and is there a way to improve this?

Thanks,
RP

**Comment by ascillato** [comment, marked as answer]
1 and 2 are because of the same thing: Sleep time.
The HTTP response time is controlled by the sleep time. Tasmota sleeps between requests in order to use less energy. Less energy, less temperature too. If you want a faster response, go to the Tasmota console and type `sleep 0`

**Comment by ReggiePark** [comment]
@ascillato Thanks, but sleep 0 made it worse, in fact, I even got frequent http timeouts.

It currently is "SleepMode":"Dynamic","Sleep":50

**Reply by ascillato** [reply]
Can you show some logs? If you have timeouts your issue is different.

**Reply by ReggiePark** [reply]
What kind of logs do you mean, the web console output?

**Reply by ascillato2** [reply]
Yes, exactly. Please, set sleep to 0 and weblog to 4 before your tests. Please, also share the output of status 0 command. Thanks.

**Reply by ReggiePark** [reply]
Trying to use the serial connection to log everything I noticed that it gets disabled after 10 minutes. Is there a way to turn this off (couldn't find any option)? Debugging a wifi problem over wifi is not the best way to do it...

**Comment by barbudor** [comment]
Hi
This is standard behavior and there is no option to disable it
When I need a permanent serial log, I add a rule such as
```
On time#minute|300 do SerialLog 3 endon
```
Which relaunch the 10 minutes timeout every 5 minutes

**Reply by ReggiePark** [reply]
Thanks.
Since I am new to Tasmota I read up on rules and did get it to work, but used time#minute|5 instead of 300 since it's supposed to be minutes. However, it appears that the timeout of 10 minutes gets triggered only once after the reboot and just one seriallog command will keep it running until the next reboot, even if you issue it during the first 10 minutes. So, a rule may only be needed if you want to keep it running after every reboot, which may not be needed in my case...

**Reply by barbudor** [reply]
My brain was not fully awake. Indeed, 'Minute' is in minutes :)
I'm surprised by your 2nd feedback. I'll will test on my side.
